engine-precompiles/src/modexp.rs

```diff
@@ -459,38 +480,28 @@ mod tests {
 
     #[test]
     fn test_berlin_modexp_big_input() {
-        let base_len = U256::from(4);
-        let exp_len = U256::from(u64::MAX);
-        let mod_len = U256::from(4);
-        let base: u32 = 1;
-        let exp = U256::MAX;
-
-        let mut input: Vec<u8> = Vec::new();
-        input.extend(u256_to_arr(&base_len));
-        input.extend(u256_to_arr(&exp_len));
-        input.extend(u256_to_arr(&mod_len));
-        input.extend(base.to_be_bytes());
-        input.extend(u256_to_arr(&exp));
-
+        let input = generate_modexp_test_input(
```

Comment:
I think passing these argus as a struct would make it cleaner, or maybe even declarative macro where the arguments are key-prefixed. E.g.:
```rust
let input = generate_modexp_test_input(ModexpInput {
    base_len: U256::from(0x80),
    exp_len: U256::from(0x80),
    mod_len: U256::from(0x80),
    base: U256::from(0x80),
    exp: U256::from(0x80),
    modulus: U256::from(0x80),
});
```
or:
```rust
let input = modexp_test_input {
    base_len: U256::from(0x80),
    exp_len: U256::from(0x80),
    mod_len: U256::from(0x80),
    base: U256::from(0x80),
    exp: U256::from(0x80),
    modulus: U256::from(0x80),
};
```

Reply:
I think It is nice to have, however it is all about test inputs not sure why we would need to have a macro for it.

Reply:
+1 for struct as input.

I personally try to avoid macros, except where they save lots of code duplication, because the IDE support for them is not as good as normal rust syntax.

Reply:
Fair enough, struct inputs it is then.

@0x3bfc, well, tests are first class citizens so the same code hygiene should apply as with normal code. In this case the reasoning is that given all six arguments are of the same type, there is nothing stopping one from accidentally switching up some of the values--future contributors could make that mistake. With the struct, however, it's very clear what purpose a value corresponds to. As a rule of thumb, in Rust one should rely on the type system as much as possible.

The struct based approach also makes it easier to read what's going on for those who aren't intimately familiar with the code base and to get an idea what the tests are doing (which is important because tests are a form of live documentation).